tools/build_paks.py

```diff
@@ -16,7 +16,8 @@
 if __name__ == '__main__':
 
     parser = argparse.ArgumentParser()
-    parser.add_argument("--unreal_engine_dir", required=True)    
+    parser.add_argument("--unreal_engine_dir", required=True)
+    parser.add_argument("--unreal_project_dir", default=os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "cpp", "unreal_projects", "SpearSim")))
```

Comment:
I just realized that our convention here is inconsistent. Some tools (e.g., `build_executable`) assume that `unreal_project_dir` is hard-coded relative to the `tools` folder. Other tools (e.g., `clean_project_dir`) have an argument for the project dir but set its default value.

I don't have a strong opinion about which convention we use, but let's use this PR as an opportunity to make it consistent across all the tools. Hard-coding is simpler and we only expect to have one project folder for the foreseeable future, but having an argument is probably more user-friendly because then users with their own projects can use our tools more easily. I think I like the convention you're using here, but let's apply it consistently to all the tools.

Reply:
I agree with this. I've made other files adhere to the convention in this file.
`build_executable` is a bit complicated as it is designed to build an executable after cloning from git repo, so as to assign a default project dir will incur more `if` statements and make the code a bit ugly. On the positive side, this script can be used to build executables from a working project dir, and this would be a bit more general purpose.

What do you think?

Reply:
Update: Actually this isn't as inconsistent as I thought. Currently, our policy seems to be, do it the hard-coded way unless there is a good reason not to. In the case of the scripts that are referred to in `build_executable`, we need to make them parameterized, because of the temp folder that is used in `build_executable`.

Reply:
Speaking of which, there is one thing that always bothered me with our approach for building pak files. There is some redundancy between the pak file that gets built by default with the executable, and the scene-specific pak files.

If it is easy to do so, we should remove all files from the scene-specific pak manifest that are already in the default manifest. Since running `build_executable` is a prerequisite for running `build_paks`, there should be the default pak manifest should be lying around somewhere in the project folder. Can we diff each scene-specific pak manifest against the default manifest so our pak files are smaller?

Reply:
You are right, there are redundancies that we can avoid in building `pak` files. Let me check on this and get back.